### ISSUE TITLE
A recent C++ Interop pull broke cross-compiling the stdlib with an alternate sysroot for SwiftGlibc platforms

### ISSUE THREAD

**Issue by buttaface** [apple/swift#60224]
**Describe the bug**
My Android CI builds each new snapshot tag of the stdlib with the Android NDK, but since #59846 it cannot [cross-compile the stdlib](https://github.com/buttaface/swift-android-sdk/runs/7496710224?check_suite_focus=true#step:6:1137):
```
swift/stdlib/public/Platform/Glibc.swift.gyb:13:19: error: no such module 'SwiftGlibc'
@_exported import SwiftGlibc // Clang module
```
**Steps To Reproduce**
1. Cross-compile the stdlib with an alternate libc sysroot.

**Expected behavior**
Previously, the Swift compiler would [look in the `-sdk` and `-resource-dir` flags passed in for a Swift resource directory](https://github.com/apple/swift/pull/59846/files#diff-f768ac8c03b01ed7761b76d70eaa046bd6032a4b7559873d13fd46a5033c9e83R35) and use the `glibc.modulemap` from there. Now, it [only remaps any Include directories the clang Driver is aware of](https://github.com/apple/swift/pull/59846/files#diff-f768ac8c03b01ed7761b76d70eaa046bd6032a4b7559873d13fd46a5033c9e83R183), which works fine if using the system libc sysroot,  but not for alternate cross-compilation sysroots. 

**Environment**
 - OS: Ubuntu 20.04 x86_64 building for Android API 24 with LTS NDK 23c
 
**Additional context**
@3405691582 has also noted that [a mistaken check from that pull broke building for non-Linux platforms like BSD](https://github.com/apple/swift/pull/59846/files#diff-f768ac8c03b01ed7761b76d70eaa046bd6032a4b7559873d13fd46a5033c9e83R172).

@egorzhdan, let me know if you have a fix in mind.

**Comment by buttaface** [comment]
@colemancda, maybe you can reproduce with your Linux armv7 sysroot, if you build with trunk?

**Comment by egorzhdan** [comment]
I think it's possible that we're not passing `-target-triple` to Clang driver when querying the include dirs.

**Comment by colemancda** [comment]
@buttaface I'll test sometime this week with my [Debian 11 (armhf) cross compilation scripts](https://github.com/colemancda/swift-armv7).

**Comment by buttaface** [comment]
> I think it's possible that we're not passing -target-triple to Clang driver when querying the include dirs.

What difference do you believe that would make?

I just [dumped `parsedIncludeArgs` natively on Android](https://github.com/apple/swift/pull/59846/files#diff-f768ac8c03b01ed7761b76d70eaa046bd6032a4b7559873d13fd46a5033c9e83R183), where your injection pull works fine, and it listed five Include directories: the first was the newly built clang resource directory at `llvm-android-aarch64/lib/clang/13.0.0/include` and the remaining four were directories in the Termux sysroot at `/data/data/com.termux/files/`. That's only possible because [I build the Swift clang with that Termux default sysroot passed in when clang is built](https://github.com/apple/swift/blob/b5735559ebfcd3d5d0672dceba05dd46922f2763/utils/build-script-impl#L533), so the clang Driver finds the Termux sysroot fine.

Obviously, that's not going to work for arbitrary libc sysroots that are passed in when cross-compiling the Swift stdlib with a prebuilt Swift compiler.

**Comment by buttaface** [comment]
@colemancda, try it yet?

This also [broke the community Android CI](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android/1412/console). I'll try dumping what paths the pull that broke this is using when cross-compiling from linux to Android.

**Comment by 3405691582** [comment]
I'm also noticing this now I'm starting to run some cross-compiling experiments.

**Comment by buttaface** [comment]
It still worked with the June 26 trunk snapshot before #59846 was merged, so [I pinned my Android CI to that tag](https://github.com/buttaface/swift-android-sdk/commit/7dfbe0500789d2f504f1984e531de787ec7cd069#diff-76992bc322b65ab94f299c05ba3cbd8b3ab2e32ebfdde6db5adba70fa9f5a26aR30). I'm going to experiment with a fix for this issue in the next couple days, but anyone can feel free to beat me to it.

**Comment by buttaface** [comment]
I built the Aug. 6 trunk source snapshot on linux x86_64 last week and dumped `parsedIncludeArgs` there too and just as I suspected it is only searching the default `/` sysroot for libc headers, even though the Android stdlib cross-compilation build passes in the NDK sysroot to `-sdk`.

I just looked and I had not noticed that [Egor had attempted to pass such alternative sysroot paths to the clang Driver](https://github.com/apple/swift/blob/6a55161187cebb129326a07d458862d1a47ec2dd/lib/ClangImporter/ClangIncludePaths.cpp#L166), but for whatever reason that is obviously not working. I will dump some more of those paths and try to figure out why.
